src/d3a/models/strategy/update_frequency.py

```diff
@@ -30,24 +30,48 @@ def __init__(self, initial_rate, final_rate, fit_to_limit=True,
                     minutes=ConstSettings.GeneralSettings.DEFAULT_UPDATE_INTERVAL),
                  rate_limit_object=max):
         self.fit_to_limit = fit_to_limit
-        self.initial_rate_profile_buffer = read_arbitrary_profile(InputProfileTypes.IDENTITY,
-                                                                  initial_rate)
+        self.initial_rate_input = initial_rate
+        self.final_rate_input = final_rate
+        self.energy_rate_change_per_update_input = energy_rate_change_per_update
+        self.initial_rate_profile_buffer = {}
+        self.final_rate_profile_buffer = {}
         self.initial_rate = {}
-        self.final_rate_profile_buffer = read_arbitrary_profile(InputProfileTypes.IDENTITY,
-                                                                final_rate)
         self.final_rate = {}
-        if fit_to_limit is False:
-            self.energy_rate_change_per_update_profile_buffer = \
-                read_arbitrary_profile(InputProfileTypes.IDENTITY, energy_rate_change_per_update)
-        else:
-            self.energy_rate_change_per_update_profile_buffer = {}
+        self.fit_to_limit = fit_to_limit
+        self.energy_rate_change_per_update_profile_buffer = {}
+        self._read_or_rotate_rate_profiles(GlobalConfig.start_date)
 
         self.energy_rate_change_per_update = {}
         self.update_interval = update_interval
         self.update_counter = {}
         self.number_of_available_updates = 0
         self.rate_limit_object = rate_limit_object
 
+    def _read_or_rotate_rate_profiles(self, current_timestamp: DateTime,
```

Comment:
A general observation by looking at the signature of this method. I think that all methods that read or rotate profiles should accept the timestamp as input and not depend on the market timeslot. That way the method will be more decoupled from its class and will be possible to abstract away more easily in a dependent class, if this needs to become more complex in the future (even now, I cannot help but have the impression that the if/else blocks that check whether a profile or a value should be used could be simplified by having a separate class instead of a method, however I am not sure it is worth it)

Reply:
`(even now, I cannot help but have the impression that the if/else blocks that check whether a profile or a value should be used could be simplified by having a separate class instead of a method, however I am not sure it is worth it)` Sorry, I dont understand what you mean. 

For the latter: Yes, this is why i left the TODO comment: `# TODO: this needs to be implemented to except profile UUIDs and DB connection`
I left it out for now in order to wrap this story up eventually. The API of the strategies and the UpdateFrequencyMixin will have to change in order to also except the price profile uuids. 
Please let me know what you think: Shall I add it here or in a separate story.?

Reply:
Just hopping in regarding the `UpdateFrequencyMixin`: I would vote for adding it in a separate story since this one is already quite big :)

Reply:
Regarding UpdateFrequencyMixin I also agree that this should be added in a new story, however keep in mind that with my d3a-web change I do not differentiate in price and energy profiles, they are all stored in the DB. Therefore if someone  uploads a price profile, it will be in the DB, and the UpdateFrequencyMixin won't be able to read it, therefore before I deploy the d3a-web changes, the to-be-created story needs to be implemented in order to not affect the users.

Regarding this `(even now, I cannot help but have the impression that the if/else blocks that check whether a profile or a value should be used could be simplified by having a separate class instead of a method, however I am not sure it is worth it)` the reason why I am stating this is because I see scattered throughout the strategy classes this check `if d3a.constants.CONNECT_TO_PROFILES_DB`. This makes me think that instead of if/else blocks, the decision for the strategy from where to retrieve the profile could be managed by a separate class, which would contain all the members of the strategy classes that can be user profiles. That way this if block (and in general profile management and decisions from where will the profile be read) would live only in the new class and not in the strategy class. I know it is probably overengineering, however the existence of all these if/else blocks is an indication that an extra abstraction layer would make the code easier to read.

Reply:
`Therefore if someone uploads a price profile, it will be in the DB` - ATM we dont have any user interface for that. I know we were discussing this in the many discussions before this PR, but really, we only support uploading the power profiled of the Load and PV and the buying and selling rate profiles of the InfiniteBusStrategy now. In order to support uploading price profiles for the template strategies, also the d3a-web will need to be changed. Or am I missing something that  you will add on the d3a-web part?

Thanks for your explanation of the latter. It makes sense and i will give it a go and try it. If it is too much hassle, I'll stop and keep the if blocks (or the mentioned function for this). OK?

Reply:
My bad, I was under the impression that we also supported price profiles for the Load / PV strategies via the UI, but I was obviously mistaken, please ignore my comment.